packages/snaps-controllers/src/services/AbstractExecutionService.ts

```diff
@@ -441,11 +441,14 @@ export abstract class AbstractExecutionService<WorkerType>
     }
 
     log('Parent: Sending Command', message);
-    const response: PendingJsonRpcResponse<Json> = await job.rpcEngine.handle(
-      message,
-    );
-
-    if (response.error) {
+    // eslint is blocking `await` usage even though `handle` returns a promise.
+    // eslint-disable-next-line @typescript-eslint/await-thenable
+    const response = await job.rpcEngine.handle(message);
+
+    if (
+      ((res): res is JsonRpcFailure =>
+        'error' in res && res.error !== undefined)(response)
+    ) {
```

Comment:
This looks a bit messy. Do we not have a `isJsonRpcError` function somewhere?

Reply:
Looks like I was reinventing the wheel here. Fixed: https://github.com/MetaMask/snaps/pull/2445/commits/3eb9330cec79f88d0e30e011fc7aa791a5216e1c